packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts

```diff
@@ -108,81 +109,34 @@ function handleNodeJSGlobals(
-		const globalMapping = inject[globalName];
-
-		if (typeof globalMapping === "string") {
-			const globalPolyfillSpecifier = globalMapping;
-
-			return {
-				contents: `
-								import globalVar from "${globalPolyfillSpecifier}";
-
-								${
-									/*
-								// ESBuild's inject doesn't actually touch globalThis, so let's do it ourselves
-								// by creating an exportable so that we can preserve the globalThis assignment if
-								// the ${globalName} was found in the app, or tree-shake it, if it wasn't
-								// see https://esbuild.github.io/api/#inject
-								*/ ""
-								}
-								const exportable =
-									${
-										/*
-									// mark this as a PURE call so it can be ignored and tree-shaken by ESBuild,
-									// when we don't detect 'process', 'global.process', or 'globalThis.process'
-									// in the app code
-									// see https://esbuild.github.io/api/#tree-shaking-and-side-effects
-									*/ ""
-									}
-									/* @__PURE__ */ (() => {
-										return globalThis.${globalName} = globalVar;
-									})();
-
-								export {
-									exportable as '${globalName}',
-									exportable as 'globalThis.${globalName}',
-								}
-							`,
-			};
-		}
-
-		const [moduleName, exportName] = inject[globalName];
+		const { importStatement, exportName } = getGlobalInject(inject[globalName]);
 
 		return {
-			contents: `
-							import { ${exportName} } from "${moduleName}";
-
-							${
-								/*
-							// ESBuild's inject doesn't actually touch globalThis, so let's do it ourselves
-							// by creating an exportable so that we can preserve the globalThis assignment if
-							// the ${globalName} was found in the app, or tree-shake it, if it wasn't
-							// see https://esbuild.github.io/api/#inject
-							*/ ""
-							}
-							const exportable =
-								${
-									/*
-								// mark this as a PURE call so it can be ignored and tree-shaken by ESBuild,
-								// when we don't detect 'process', 'global.process', or 'globalThis.process'
-								// in the app code
-								// see https://esbuild.github.io/api/#tree-shaking-and-side-effects
-								*/ ""
-								}
-								/* @__PURE__ */ (() => {
-									return globalThis.${globalName} = ${exportName};
-							})();
-
-							export {
-								exportable as '${globalName}',
-								exportable as 'global.${globalName}',
-								exportable as 'globalThis.${globalName}'
-							}
-						`,
+			contents: `${importStatement}\nglobalThis.${globalName} = ${exportName};`,
 		};
 	});
 }
 
+/**
+ * Get the import statement and export name to be used for the given global inject setting.
+ */
+function getGlobalInject(globalInject: string | string[]) {
+	if (typeof globalInject === "string") {
+		// the mapping is a simple string, indicating a default export, so the string is just the module specifier.
+		return {
+			importStatement: `import globalVar from "${globalInject}";`,
+			exportName: "globalVar",
+		};
+	} else {
```

Comment:
nit: else is not required here...

Reply:
Yeah I can never decide which I prefer. It certainly isn't needed but equally if someone decide to remove that return then it would accidentally fall through to the rest.